runtime/src/chpl-file-utils.c

```diff
@@ -164,16 +164,66 @@ qioerr chpl_fs_is_mount(int* ret, const char* name) {
   int exitStatus = 0;
   size_t nameLen = strlen(name);
   char* parent = (char* ) chpl_mem_allocMany(nameLen + 4, sizeof(char), CHPL_RT_MD_OS_LAYER_TMP_DATA, 0, 0);
-  strncpy(parent, name, nameLen + 1);
-  strncat(parent, "/..", 3);
-  // TODO: Using "/" is not necessarily portable, look into this
+  char* safeNameCopy = (char* ) chpl_mem_allocMany(nameLen + 1, sizeof(char), CHPL_RT_MD_OS_LAYER_TMP_DATA, 0, 0);
+  strncpy(safeNameCopy, name, nameLen + 1);
+  // Need to copy name so that we can use it in the case of links
 
-  exitStatus = stat(name, &nBuf);
+  err = chpl_fs_is_link(&exitStatus, name);
+  if (err) {
+    // The stat call in is_link returned an error, which we would encounter too,
+    // so return immediately.
+    return err;
```

Comment:
The strings parent and safeNameCopy don't get freed in these early returns for errors.